index.d.ts

```diff
@@ -0,0 +1,758 @@
+   * @param {Array} y infix?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if infix
+   */
+  export function isInfixOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): boolean;
+
+  /**
+   * Checks if array has a permutation.
+   * @param {Array} x an array
+   * @param {Array} y permutation?
+   * @param {function?} fn compare function (a, b)
+   * @returns {boolean} true if permutation
+   */
+  export function isPermutation<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): boolean;
+
+  /**
+   * Checks if array has a permutation.
+   * @param {Array} x an array
+   * @param {Array} y permutation?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if permutation
+   */
+  export function isPermutationOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): boolean;
+
+  /**
+   * Checks if array starts with a prefix.
+   * @param {Array} x an array
+   * @param {Array} y prefix?
+   * @param {function?} fn compare function (a, b)
+   * @returns {boolean} true if prefix
+   */
+  export function isPrefix<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): boolean;
+
+  /**
+   * Checks if array starts with a prefix.
+   * @param {Array} x an array
+   * @param {Array} y prefix?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if prefix
+   */
+  export function isPrefixOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): boolean;
+
+  /**
+   * Checks if array has a subsequence.
+   * @param {Array} x an array
+   * @param {Array} y subsequence?
+   * @param {function?} fn compare function (a, b)
+   * @returns {boolean} true if subsequence
+   */
+  export function isSubsequence<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): boolean;
+
+  /**
+   * Checks if array has a subsequence.
+   * @param {Array} x an array
+   * @param {Array} y subsequence?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if subsequence
+   */
+  export function isSubsequenceOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): boolean;
+
+  /**
+   * Checks if array ends with a suffix.
+   * @param {Array} x an array
+   * @param {Array} y suffix?
+   * @param {function?} fn compare function (a, b)
+   * @returns {boolean} true if suffix
+   */
+  export function isSuffix<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): boolean;
+
+  /**
+   * Checks if array ends with a suffix.
+   * @param {Array} x an array
+   * @param {Array} y suffix?
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {boolean} true if suffix
+   */
+  export function isSuffixOn<T>(x: T[], y, fn?: (v: T, i: number, x: T[]) => T[], ths?: object);
```

Comment:
Missed type here for `y`:

```suggestion
  export function isSuffixOn<T>(x: T[], y: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object);
```

Reply:
Oops lol, thanks for letting me know